### PR TITLE
New empty_assignment_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #884

### CODE REVIEW

**Comment by IndrajeetPatil** [comment]
These should produce lints, right?

``` r
library(lintr)

{} -> x
x
#> NULL

lint(text = "{} -> x", linters = empty_assignment_linter())
lint(text = "{} ->> x", linters = empty_assignment_linter())
```

<sup>Created on 2022-10-08 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

P.S. Funny branch name 😆

**Comment by MichaelChirico** [comment]
> These should produce lints, right?

yep; had left it as a TODO but it's a trivial addition; done

> P.S. Funny branch name

😅

File: R/empty_assignment_linter.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    xml <- source_expression$xml_parsed_content
+
+    bad_expr <- xml2::xml_find_all(xml, xpath)
+
+    xml_nodes_to_lints(
+      bad_expr,
+      source_expression = source_expression,
+      lint_message =
+        "Assign NULL explicitly or, whenever possible, allocate the empty object with the right type & size.",
```
Comment (IndrajeetPatil) [inline R/empty_assignment_linter.R:58]:
```suggestion
        "Assign NULL explicitly or, whenever possible, allocate the empty object with the right type and size.",
```

**Review by IndrajeetPatil — APPROVED** [review]
Cool stuff. Thanks!

**Review by IndrajeetPatil — APPROVED** [review]

**Pull request merged**